### PR TITLE
Shim NotAllowedError.

### PR DESCRIPTION
**Pull request by jan-ivar**
**Description**
Shim getUserMedia to fail with NotAllowedError (Chrome, Firefox) and OverconstrainedError (Chrome), to match the spec.

**Purpose**
Make it easier to handle errors across browsers and versions uniformly.

### CODE REVIEW

**Comment by jan-ivar** [comment]
Need to test Edge to see if it needs a shim as well.

File: src/js/chrome/getusermedia.js [hunk 1]
```diff
@@ -101,9 +101,21 @@ module.exports = function() {
     return func(constraints);
   };
 
+  var shimError_ = e => ({
+    name: {PermissionDeniedError: 'NotAllowedError',
+           ConstraintNotSatisfiedError: 'OverconstrainedError'
+          }[e.name] || e.name,
+    message: e.message,
+    constraint: e.constraintName,
+    toString: function() {
```
Comment (KaptenJansson) [inline src/js/chrome/getusermedia.js:110]: I'm curious to why you did not use an arrow function here?

File: src/js/firefox/getusermedia.js [hunk 2]
```diff
@@ -112,4 +129,10 @@ module.exports = function() {
       });
     };
   }
+  if (browserDetails.version < 49) {
+    var origGetUserMedia = navigator.mediaDevices.getUserMedia.
```
Comment (KaptenJansson) [inline src/js/firefox/getusermedia.js:133]: I would rather have the dot on the new line but since we do not have linting in place for this I will fix it afterwards when I add the new [rule](http://eslint.org/docs/rules/dot-location).

**Comment by KaptenJansson** [comment]
LGTM

**Comment by fippo** [comment]
@jan-ivar: what Edge version did you test with? Might be worth fixing upstream, @shijuns might be able to arrange this ;-)

File: src/js/chrome/getusermedia.js (hunk 1 above)
Comment (jan-ivar) [inline src/js/chrome/getusermedia.js:110]: I didn't want to assume strict mode. http://stackoverflow.com/questions/36717376/arrow-function-in-object-literal

Are we guaranteeing adapter.js will always use strict?

**Comment by jan-ivar** [comment]
I tested Edge 25.10586.0.0, which appears to use `PermissionDeniedError`.

**Comment by ShijunS** [comment]
@fippo, @Jan-Ivar, We haven't made much change to the Edge implementation recently.  Now with the spec being finalized soon, we will catch up in the next product cycles in a few areas including the error event and error code.

File: src/js/chrome/getusermedia.js (hunk 1 above)
Comment (KaptenJansson) [inline src/js/chrome/getusermedia.js:110]: I'm pretty sure our linters (and browsers) will let us know if we do not follow strict conventions since we have 'use strict' at the top of each file.
